src/components/RichTextEditor/utils/parseRawContent.ts

```diff
@@ -0,0 +1,77 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Descendant } from "slate";
+import { jsx } from "slate-hyperscript";
+import { BlockStyleTypes } from "../renderer/renderBlockStyles";
+import { Styles } from "../renderer/renderInlineStyles";
+
+const BLOCK_MAP: { [key: string]: (el: HTMLElement) => { type: string } } = {
+    A: (el) => ({ type: BlockStyleTypes.Link, url: el.getAttribute("href") }),
+    P: () => ({ type: BlockStyleTypes.Paragraph }),
+    PRE: () => ({ type: BlockStyleTypes.Code }),
+    UL: () => ({ type: BlockStyleTypes.UnorderedList }),
+    OL: () => ({ type: BlockStyleTypes.OrderedList }),
+    LI: () => ({ type: BlockStyleTypes.ListItem }),
+};
+
+const STYLE_MAP: { [key: string]: () => { [key: string]: boolean } } = {
+    I: () => ({ [Styles.Italic]: true }),
+    S: () => ({ [Styles.Strikethrough]: true }),
+    STRONG: () => ({ [Styles.Bold]: true }),
+    BOLD: () => ({ [Styles.Bold]: true }),
+    B: () => ({ [Styles.Bold]: true }),
+    U: () => ({ [Styles.Underline]: true }),
+};
+
+export const parseRawValue = (raw?: string): Descendant[] => {
+    let parsedValue: Descendant[] = [{ type: BlockStyleTypes.Paragraph, children: [{ text: "" }] }];
+
+    if (!raw) {
+        return parsedValue;
+    }
+
+    try {
+        parsedValue = JSON.parse(raw);
+    } catch {
+        const parsedDoc = new DOMParser().parseFromString(raw, "text/html");
+        const parsedHtml = deserializeHTML(parsedDoc.body) as Descendant[];
+        if (parsedHtml) {
+            parsedValue = parsedHtml;
+        }
+    }
+
+    return parsedValue;
+};
+
+const deserializeHTML = (el: HTMLElement | ChildNode): Descendant | Descendant[] | string | null => {
+    if (el.nodeType === 3) {
+        return el.textContent;
+    } else if (el.nodeType !== 1) {
+        return null;
+    }
+    let children = Array.from(el.childNodes)
+        .filter((el) => (el.textContent ? el.textContent.trim().length > 0 : false))
```

Comment:
```suggestion
        .filter((el) => (el.textContent?.trim().length > 0))
```

Reply:
this will complain that the object is possibly undefined